helix-term/src/ui/editor.rs

```diff
@@ -488,24 +495,41 @@ impl EditorView {
             );
         }
 
-        surface.set_stringn(
-            viewport.x + viewport.width.saturating_sub(15),
-            viewport.y,
-            format!("{}", doc.diagnostics().len()),
-            4,
-            text_color,
-        );
-
-        // render line:col
-        let pos = coords_at_pos(doc.text().slice(..), doc.selection(view.id).cursor());
-
-        let text = format!("{}:{}", pos.row + 1, pos.col + 1); // convert to 1-indexing
-        let len = text.len();
+        //-------------------------------
+        // Right side of the status line.
+        //-------------------------------
+
+        // Compute the individual info strings.
+        let diag_count = format!("{}", doc.diagnostics().len());
+        let indent_info = match doc.indent_style {
+            IndentStyle::Tabs => "tab",
+            IndentStyle::Spaces(1) => "spaces:1",
+            IndentStyle::Spaces(2) => "spaces:2",
+            IndentStyle::Spaces(3) => "spaces:3",
+            IndentStyle::Spaces(4) => "spaces:4",
+            IndentStyle::Spaces(5) => "spaces:5",
+            IndentStyle::Spaces(6) => "spaces:6",
+            IndentStyle::Spaces(7) => "spaces:7",
+            IndentStyle::Spaces(8) => "spaces:8",
+            _ => "indent:ERROR",
+        };
```

Comment:
Hmm, I'm not used to having this on the status line, instead using commands to look this up. I think we could keep it, but in order to save space on the status line, we only display it if it's different from the language default. WDYT?

I do a similar thing for the encoding (only display if not UTF-8) and line ending in my vim setup.

Reply:
I guess I'm kind of a dashboard person, ha ha.  Personally, I would definitely prefer both the encoding and indentation style to be shown at all times.  But I'm probably weird.  (Also, I'll probably get used to it not being there anyway.)

Would you be okay with just commenting out the display code for this PR, and then we can revisit that as part of a more general discussion of how/when/what information is displayed on the status line?

Reply:
Will comment it out, as discussed on Matrix.